Mage.Sets/src/mage/cards/d/DaghatarTheAdamant.java

```diff
@@ -82,19 +79,14 @@ public MoveCounterFromTargetToTargetEffect copy() {
 
     @Override
     public boolean apply(Game game, Ability source) {
-        Player controller = game.getPlayer(source.getControllerId());
-        MageObject sourceObject = game.getObject(source);
-        if (sourceObject != null && controller != null) {
-            Permanent fromPermanent = game.getPermanent(getTargetPointer().getFirst(game, source));
-            if (fromPermanent != null && fromPermanent.getCounters(game).getCount(CounterType.P1P1) > 0) {
-                Permanent toPermanent = game.getPermanent(source.getTargets().get(1).getFirstTarget());
-                if (toPermanent != null) {
-                    fromPermanent.removeCounters(CounterType.P1P1.createInstance(), source, game);
-                    toPermanent.addCounters(CounterType.P1P1.createInstance(), source.getControllerId(), source, game);
-                    game.informPlayers(sourceObject.getLogName() + ": Moved a +1/+1 counter from " + fromPermanent.getLogName() +" to " + toPermanent.getLogName());
-                }
+        Permanent fromPermanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+        if (fromPermanent != null && fromPermanent.getCounters(game).getCount(CounterType.P1P1) > 0) {
+            Permanent toPermanent = game.getPermanent(source.getTargets().get(1).getFirstTarget());
+            if (toPermanent != null) {
+                fromPermanent.removeCounters(CounterType.P1P1.createInstance(), source, game);
+                toPermanent.addCounters(CounterType.P1P1.createInstance(), source.getControllerId(), source, game);
+                return true;
```

Comment:
Shouldn't this return true only if (at least) one of the `removeCounters`/`addCounters` calls returns true?

Reply:
`removeCounters` doesn't have a return value but I changed this to return the value it gets back form `addCounters`.